person.rb

```diff
@@ -1,35 +1,19 @@
-require_relative 'nameable'
-
-# Represents a person with attributes such as name, age, and permission.
-class Person < Nameable
+class Person
   attr_accessor :name, :age
   attr_reader :rentals
 
+  @all = []
+
+  class << self
+    attr_reader :all
+  end
+
   def initialize(age, name: 'Unknown', parent_permission: true)
-    super()
     @name = name
     @age = age
     @parent_permission = parent_permission
     @rentals = []
-  end
-
-  def correct_name
-    name
-  end
-
-  def can_use_services?
-    of_age? || @parent_permission
-  end
-
-  def add_rental(book, date)
-    rental = Rental.new(date, book, self)
-    @rentals << rental
-    book.rentals << rental
-  end
-
-  private
 
-  def of_age?
-    age >= 18
+    self.class.all << self
```

Comment:
- [ ] [REQUIRED] Good job so far! Kindly I can't create every type of persons(`Teacher`, `Student`) because you have got error with this part therefore I couldn't check your `rentals` part as mentioned previous reviewer, please fix it 🙏🏼 Below I put pictures about error `=>`

- `Student`
![image](https://github.com/Bellagirl-maker/oop-school-library/assets/63915024/4585ab31-a6a2-48e2-85d2-78316c784e65)

- `Teacher`
![image](https://github.com/Bellagirl-maker/oop-school-library/assets/63915024/7e98dd73-1e54-4677-aa2e-9f3140a255a9)